magento/magento2
# Support batches processing for media gallery synchronization queue messages

### Description (*)
Added an implementation to support batch processing for synchronization queue messages

### Related Pull Requests
https://github.com/magento/partners-magento2-infrastructure/pull/36


### Fixed Issues (if relevant)
1. Fixes magento/adobe-stock-integration#1724: Support batches processing for synchronization queue messages

### Manual testing scenarios (*)

Check synchronization when enabling new media gallery
Check `bin/magento media-gallery:sync` functionality
Check `bin/magento media-content:sync` functionality

General synchronization test case:

1. Upload a couple of images or save images from Adobe Stock
2. Add/insert images to either category, product, cms pages or blocks
3. Delete rows from tables `media_gallery_asset`  and `media_content_asset`
4. Disable Enhanced Media Gallery and save configuration `(Assuming it is already enabled)`
5. Enable Enhanced Media Gallery and save configuration `(Changes in the system configuration for the Enhanced Media Gallery triggers the Media Gallery and Media Content sychronization publishers)`
6. Run `bin/magento queue:consumers:start media.gallery.synchronization`
7. Open Media Gallery and ensure the images are present in the grid. (Check `media_gallery_asset` table if data are saved)
8. Run `bin/magento queue:consumers:start media.content.synchronization`
9. View details of the images in the media gallery and ensure the Used In information is correct. (Check `media_content_asset` table if data are saved)

Batch synchronization test case:
1. Upload a couple of images or save images from Adobe Stock
2. Add/insert these images to either category, product, cms pages or blocks
3. Delete rows from tables `media_gallery_asset`  and `media_content_asset`
4. Call the `\Magento\MediaGallerySynchronization\Model\Publish::execute` passing the paths of the images
5. Run `bin/magento queue:consumers:start media.gallery.synchronization`
6. Open Media Gallery and ensure the images are present in the grid. (Check `media_gallery_asset` table if data are saved)
7. Call the `\Magento\MediaContentSynchronization\Model\Publish::execute` passing the contentIdentities of the places where the images have been added to
8. Run `bin/magento queue:consumers:start media.content.synchronization`
9. View details of the images in the media gallery and ensure the Used In information is correct. (Check `media_content_asset` table if data are saved)

### Questions or comments

### Contribution checklist (*)
